src/Components/Common/SearchByMultipleFields.tsx

```diff
@@ -0,0 +1,220 @@
+  options,
+  onSearch,
+  initialOption,
+  className,
+  inputClassName,
+  buttonClassName,
+}) => {
+  const { t } = useTranslation();
+  const [selectedOption, setSelectedOption] = useState<SearchOption>(
+    initialOption || options[0],
+  );
+  const [searchValue, setSearchValue] = useState(selectedOption.value || "");
+  const [open, setOpen] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === "/" && document.activeElement !== inputRef.current) {
+        e.preventDefault();
+        setOpen(true);
+      }
+
+      if (open) {
+        if (e.key === "Escape") {
+          setOpen(false);
+        }
+      }
+
+      options.forEach((option) => {
+        if (e.key.toLowerCase() === option.shortcut_key.toLowerCase()) {
+          e.preventDefault();
+          handleOptionChange(option);
+        }
+      });
+    };
+
+    document.addEventListener("keydown", handleKeyDown);
+    return () => document.removeEventListener("keydown", handleKeyDown);
+  }, []);
+
+  const handleOptionChange = useCallback(
+    (option: SearchOption) => {
+      setSelectedOption(option);
+      setSearchValue(option.value || "");
+      setOpen(false);
+      inputRef.current?.focus();
+      onSearch(option.key, option.value);
+    },
+    [onSearch],
+  );
+
+  const handleSearchChange = useCallback(
+    (value: string) => {
+      setSearchValue(value);
+      onSearch(selectedOption.key, value);
+    },
+    [selectedOption, onSearch],
+  );
+
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      console.log(e.key);
+      if (e.key === "Escape") {
+        setSearchValue("");
+        onSearch(selectedOption.key, "");
+      }
+      if (e.key === "/") {
+        e.preventDefault();
+        setOpen(true);
+      }
+    },
+    [selectedOption, onSearch],
+  );
+
+  const renderSearchInput = useMemo(() => {
+    const commonProps = {
+      ref: inputRef,
+      value: searchValue,
+      onChange: (e: any) =>
+        handleSearchChange(e.target ? e.target.value : e.value),
+      onKeyDown: handleKeyDown,
+      className: cn(
+        "flex-grow border-none shadow-none focus-visible:ring-0 h-10",
+        inputClassName,
+      ),
+    };
+
+    switch (selectedOption.type) {
+      case "phone":
+        return (
+          <PhoneNumberFormField
+            name={selectedOption.key}
+            placeholder={t(selectedOption.placeholder)}
+            types={["mobile", "landline"]}
+            {...commonProps}
+          />
+        );
+      default:
+        return (
+          <Input
+            type="text"
+            placeholder={t(selectedOption.placeholder)}
+            {...commonProps}
+          />
+        );
+    }
+  }, [
+    selectedOption,
+    searchValue,
+    handleSearchChange,
+    handleKeyDown,
+    t,
+    inputClassName,
+  ]);
+
+  return (
```

Comment:
This component should have a default state where it says a simple help text. EG 
<img width="949" alt="image" src="https://github.com/user-attachments/assets/6ef1d574-81d2-4f84-a712-45b268f265c8">